### DISCUSSION TITLE
Google Driveに保存した画像を埋め込むと表示されたりされなかったりする

### DISCUSSION THREAD

**Discussion by ozaeri** [increments/qiita-discussions#369] [要望・アイデア・不具合]
お世話になります。

過去に同じ課題が投稿されていますが、Closeになってしまっているようなので再度投稿させていただきます。
ref: #333

Google Driveに保存した画像を一般公開設定（リンク取得）し、Qiitaに埋め込みをすると、画像が表示されたり、されなかったりします。画像がうまく表示されてない状態でもリンクをクリックすると画像自体は表示されます。公開時すぐは表示されるようになっていたのですが、本日確認したところ、表示されていない部分が多く存在しています。
一例ですが、表示されていない画像（キャプチャ中の000001.jpg）のURLは https://drive.google.com/uc?export=view&id=1DGZJWJ2TYhtLd_iftjr-Hlv13OzK4vsQ を設定しています。
<img width="1510" alt="スクリーンショット 2022-12-06 16 52 21" src="https://user-images.githubusercontent.com/29727155/205853111-d29722ee-0081-4646-a512-250f5a2b7d09.png">

画像のキャッシュサーバ（como？）が何か影響しているように見えます（comoとGoogle Driveの相性？）。
すみませんが、調査お願いします。
必要な情報があれば提供させていただきます。

**Comment by kyntk** [comment]
@ozaeri さん
ご不便をおかけして申し訳ありません！
軽く調査をしてみたのですが、CDNやプロキシ用のサーバーからGoogle Driveの画像が取得できないことがあるようです。
現時点では原因特定まではできておらず、追加調査・対応に時間がかかってしまいそうです。:bow: 

https://drive.google.com/uc?export=view&id=1DGZJWJ2TYhtLd_iftjr-Hlv13OzK4vsQ にアクセスをしたときにリダイレクトされるようですが、それが関係している可能性もあるのでは？と思っています。
Qiita上で画像が見える時と、見えない時で、上記URLのリダイレクト先のURLに何か変化があるかを確認いただくことは可能でしょうか？

また、すぐに解決できない可能性があるので、Qiita上に画像をアップロードすることで回避いただきたいです :pray:

**Reply by ozaeri** [reply]
> Qiita上で画像が見える時と、見えない時で、上記URLのリダイレクト先のURLに何か変化があるかを確認いただくことは可能でしょうか？

リダイレクト先のURLを確認したところ、見える時・見えない時で異なっていました。
今現在、同じ画像でもPCでは表示されており、モバイルでは表示されていない状況なのですが、それぞれのリダイレクトURLが以下のようになっています。

- 見える時（PCで表示できている）https://doc-04-8k-docs.googleusercontent.com/docs/securesc/cbapuki1v03f7s112m36qn76n9l3f6g3/os062krt2g40l4ogrb2mj9p261ctbasj/1670462100000/01050089750389882081/05934298933683073818/1DGZJWJ2TYhtLd_iftjr-Hlv13OzK4vsQ?e=view&ax=AEKYgyRZ0sKYMvdxbodZ8geL8QdEBOrrsPyGRINSfUTzZcYUwb_LlraCrUjJeRDvylIBgYEiWNsDNQb3rhZrzmBr0fzc5UZMLi-yQEw0iqtrYfC4YKFHKY-ZA4QYiZeGBeNcyOsr5qBUrIuze2SOvLZCN05At6AnMoY5Djjix8FObM7GSYmhEDgZpA6dE4BTwPCea9F4rnCMv4KG8hdlv_-3IeCH81uyvLRCg9UQvkbowzSLpMMarII9CLsdOj5OtFrpp2fbCUdt-3vJ16x3aCMPsyLd7bP77mhnhPMaiCr8RaGzxsr7ZeQYaftqLZG7fNTUbgUSLFOiYmrGKX7QvnQUEvuQB9FVQQBthSYRWDa0oymhYCr_3l3mNCZ6Rpiz2Ab6Ibvp9KbHL-VfETl4KLACt5SA9gd0FNFP5cXz9kHLWWp10mt95jv5APO_RNizL-SkyUmzriLSS_Y-H7ReTLrcSvYZlTTlmplCoTAavU4h69MzEcBVpU6USOXSm9iILf94KryaOadAqZCiL2TmlV3ZSbijSTV68S7AtT2dBbgzCymbYS8DLQjrCzmOWLhscKPd5YOD8ZQogkSV9ZqQXLmb3sJKTvcNyZU1zvEXG1GrtQStETprZrMzhxxPP4QKqliWEuRIAzVPZcQ6_WiAbUCnA4XYu09_Sg1J3n6m_VRoFcYaTf6UfyqcUgCVQ0PybfEBa5FYgYlSTatU5NnYhKKpdlSp0ThB8aX1oJzHmE95tt3TkLJZGGIwjNi9guZMRQbVzD6YgdsiuuDLD9r1-ff53w6wH2gYjL9OsgT1nvPvm2pYfeyMujkE7h5FqZ0u4DruvOV7dM6g9mSBaK1llUmIR2u3oKP6uw3mVs7EQcCNK268vN87yst8emXwylBtxjwVcppO&uuid=8a9ec9d7-c25a-4ea4-a0b8-c8d273169760&authuser=0
- 見えない時（モバイルで表示されない）https://camo.qiitausercontent.com/e535aef28f840b9178597f824c02481e825694b9/68747470733a2f2f64726976652e676f6f676c652e636f6d2f75633f6578706f72743d766965772669643d3144475a4a574a32545968744c645f6966746a722d486c7631334f7a4b34767351

また、private URLだとこの事象が起こりやすいように感じ、private URLで見えなかった画像も全体公開すると表示されるようになりますが、日や利用する端末によってまた見えたり見えなかったりしているようです。
（実は、最初はQiitaに上げていたのですが上限に引っかかってしまって・・・）すみませんが、よろしくお願いいたします🙏

**Reply by kyntk** [reply]
@ozaeri 追加の情報ありがとうございます！
頂いた1つ目のURLでは画像が確認できないのですが、もしかしたらGoogle Drive側で表示URLが変わったりするのでしょうかね?
Google Driveの仕様も影響していそうで、こちらですぐに解決は難しそうです... 🙇 

Qiitaの画像アップロードの容量制限については、月単位での制限になっております。
そのため、時間を空けてアップロードいただければ解消します。
お手数おかけして申し訳ありませんが、ひとまずこういった対応をお願いします 🙇 
https://help.qiita.com/ja/articles/qiita-image-upload

**Reply by ozaeri** [reply]
Google Drive側のURLですが、どうやら変わっているようです（日毎？に違うようでした）。
そして今現在の状態ですが、PC・モバイルともに画像が表示できている状態で、画像のリダイレクト先URLが追加情報記載時に「見えない時」でお伝えしていたURLでした（今はPC/モバイルどちらもGoogle Drive側のURLにはリダイレクトされていません）。

追記：別の限定記事でリンクしているものに関しては、画像が表示されず、https://camo.qiitausercontent.com/26625fe92d94e6988f21b22ccf42f218893981b1/68747470733a2f2f64726976652e676f6f676c652e636f6d2f75633f6578706f72743d766965772669643d31566361586f5168365272474c515a2d6e57345173496b5937776678796a764462 にリダイレクトされ「Not Found」と表示されています。
（記事で設定している[画像リンク](https://drive.google.com/uc?export=view&id=1VcaXoQh6RrGLQZ-nW4QsIkY7wfxyjvDb)は存在すること確認済み）

**Reply by kyntk** [reply]
@ozaeri 追加の情報をいただきありがとうございます！

ただ申し訳ないのですが、外部のサービスというのもあり、すぐに解決するのは難しそうと考えております。
そのため、対応については他の施策との優先度の兼ね合いですぐにはできなそうです。
ご理解いただけますと幸いです:bow:
